service/worker/scheduler/workflow.go

```diff
@@ -1002,6 +1004,39 @@ func (s *scheduler) getListInfo(inWorkflowContext bool) *schedpb.ScheduleListInf
 	}
 }
 
+func (s *scheduler) updateCustomSearchAttributes(searchAttributes *commonpb.SearchAttributes) {
+	if searchAttributes == nil {
```

Comment:
maybe add a comment that we explicitly want to support non-nil `searchAttributes` with empty map to mean "unset all"